File: B12 Bit Manipulation/Leetcode 461 Hamming Distance/Leetcode 461 Hamming distance.py
###########
Time Complexity:
Space Complexity:


#############
class Solution:
    def hammingDistance(self, x: int, y: int) -> int:
        count = 0
		# XOR both the numbers and then simple count the number of set bits in the result.
        x ^= y
        while x:
            if x & 1:
                count += 1
            x >>= 1
        return count
        
 ###############
 

File: B12 Bit Manipulation/Leetcode 461 Hamming Distance/Leetcode 461 Hamming Distance.java
/*

Time Complexity: O(N)
Space Complexity: O(1)
*/ 

class Solution {
    public int hammingDistance(int x, int y) {
        int cnt = 0; 
        while(x>0 || y>0){
            cnt += (x%2)^(y%2);
            x >>= 1; 
            y >>= 1;
        }
        return cnt; 
        
    }
}


##########
class Solution {
    public int hammingDistance(int x, int y) {
        int xor = x^y;
        int ans = 0;
        while(xor > 0)
        {
            if(xor%2 == 1)
                ans++;
            xor = xor/2;
        }
        return ans;
    }
}